src/bucket/BucketList.cpp

```diff
@@ -63,7 +63,7 @@ BucketLevel::getNext()
 void
 BucketLevel::setNext(FutureBucket const& fb)
 {
-    releaseAssert(threadIsMain());
+    assertThreadIsMain();
```

Comment:
note that this with change we're actually losing visibility into file + line number in case of a crash, which makes debugging much harder. This is something I faced in https://github.com/stellar/stellar-core/pull/4258 as well, so I'd actually recommend switching all `assertThreadIsMain` to `releaseAssert(threadIsMain())`